### PR TITLE
feat: add the workspace-create base functionality

### PR DESCRIPTION
**Pull request by RedCrazyGhost**
### Description

add the workspace-create base functionality

Fixes # (issue numbers)

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [X] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible

### CODE REVIEW

**Review by ZhaoKang0610 — COMMENTED** [review]
No big problem, it would be better to use validate

**Review by zccabb — COMMENTED** [review]
add test.

**Review by yuluo-yx — APPROVED** [review]
Plsa adjust code by comment.

**Review by zccabb — APPROVED** [review]

File: apps/workbench/backend/service/workspace/workspace.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+		Id:          workspaceDao.Id,
+		Name:        workspaceDao.Name,
+		Lable:       workspaceDao.Lable,
+		Description: workspaceDao.Description,
+		Logo:        workspaceDao.Logo,
+	}
+}
+
+// hashTop6 hash
+func hashTop6(str string) string {
```
[the remark this answers is not in the document]
Comment (RedCrazyGhost) [reply apps/workbench/backend/service/workspace/workspace.go:90]: I think this method of drawing out is necessary

File: apps/workbench/backend/api/workspace/workspace.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
+package workspace
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/deamgo/workbench/context"
+	"github.com/deamgo/workbench/pkg/e"
+	"github.com/deamgo/workbench/pkg/types"
+	"github.com/deamgo/workbench/service/workspace"
```
Comment (yuluo-yx) [inline apps/workbench/backend/api/workspace/workspace.go:10]: add blank line.

File: apps/workbench/backend/cmd/main.go [hunk 3]
```diff
@@ -1,12 +1,14 @@
 package main
 
 import (
+	workspace2 "github.com/deamgo/workbench/dao/workspace"
 	"github.com/deamgo/workbench/initialize"
 	"github.com/deamgo/workbench/service/mail"
+	"github.com/deamgo/workbench/service/workspace"
 	"github.com/gin-gonic/gin"
```
Comment (yuluo-yx) [inline apps/workbench/backend/cmd/main.go:8]: adjust the order pkg.

File: apps/workbench/backend/db/sql/developer_workspace.sql [hunk 4]
```diff
@@ -0,0 +1,16 @@
+create table if not exists workbench.`workspace_developer_relation`(
+   `developer_id` bigint not null comment '开发者ID',
```
Comment (yuluo-yx) [inline apps/workbench/backend/db/sql/developer_workspace.sql:2]: pls change to en comment.

File: apps/workbench/backend/router/router.go [hunk 5]
```diff
@@ -1,6 +1,7 @@
 package router
 
 import (
+	"github.com/deamgo/workbench/api/workspace"
```
Comment (yuluo-yx) [inline apps/workbench/backend/router/router.go:4]: adjust the order pkg

File: apps/workbench/backend/service/workspace/workspace.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (yuluo-yx) [reply apps/workbench/backend/service/workspace/workspace.go:90]: > I think this method of drawing out is necessary

okay.

File: apps/workbench/backend/service/workspace/workspace_test.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
+package workspace
+
+import (
+	"context"
```
Comment (yuluo-yx) [inline apps/workbench/backend/service/workspace/workspace_test.go:4]: Same as above

**Review by yuluo-yx — CHANGES_REQUESTED** [review]

**Review by yuluo-yx — APPROVED** [review]
LGTM.

**Pull request merged**